lib/vnet/admin_process_darwin.go

```diff
@@ -113,7 +117,29 @@ func RunDarwinAdminProcess(ctx context.Context, config daemon.Config) error {
 			}
 		}
 	})
-	return trace.Wrap(g.Wait(), "running VNet admin process")
+
+	done := make(chan error)
+	go func() {
+		done <- g.Wait()
+	}()
+
+	select {
+	case err := <-done:
+		return trace.Wrap(err, "running VNet admin process")
+	case <-ctx.Done():
+		timer := time.NewTimer(5 * time.Second)
+		defer timer.Stop()
+
+		select {
+		case <-done:
+			// network stack exited cleanly within timeout
+			return trace.Wrap(err, "running VNet admin process")
+		case <-timer.C:
+			log.ErrorContext(ctx, "VNet admin process did not exit within 5 seconds, forcing shutdown.")
+			os.Exit(1)
+			return nil
+		}
+	}
```

Comment:
make sure to set `err` in the second select where you receive from the channel, you can also pull the second select out to reduce indentation
```suggestion
	case <-ctx.Done():
	}

	select {
	case err := <-done:
		// network stack exited cleanly within timeout
		return trace.Wrap(err, "running VNet admin process")
	case <-time.After(5*time.Second):
		log.ErrorContext(ctx, "VNet admin process did not exit within 5 seconds, forcing shutdown.")
		os.Exit(1)
		return nil
	}
```